copy/copy.go

```diff
@@ -1069,6 +1069,19 @@ type diffIDResult struct {
 // copyLayer copies a layer with srcInfo (with known Digest and Annotations and possibly known Size) in src to dest, perhaps (de/re/)compressing it,
 // and returns a complete blobInfo of the copied layer, and a value for LayerDiffIDs if diffIDIsNeeded
 func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, toEncrypt bool, pool *mpb.Progress) (types.BlobInfo, digest.Digest, error) {
+	// If the srcInfo doesn't contain compression information, try to compute it from the
+	// MediaType, which was usually read from a manifest by way of LayerInfos(), so that
+	// we can use the information to ensure that the eventual call to UpdatedImage() that
+	// uses the blobInfo we return can correctly compute updated MediaType values.
```

Comment:
```suggestion
	// uses the blobInfo we return can correctly compute updated MediaType values
	// (sadly UpdatedImage() is documented to not update MediaTypes
	// from ManifestUpdateOptions.LayerInfos[].MediaType, so we are doing it
	// indirectly).
	//
	// This MIME type → compression mapping belongs in manifest-specific code in c/image/manifest
	// (but we should preferably replace/change UpdatedImage instead of productizing this workaround).
```

Reply:
Added.